cadc-access-control-server/src/main/java/ca/nrc/cadc/ac/server/web/LoginServlet.java

```diff
@@ -185,6 +186,18 @@ public void doPost(HttpServletRequest request, HttpServletResponse response)
             {
                 String token = null;
                 HttpPrincipal p = new HttpPrincipal(userID, proxyUser);
+
+                // Get set of all principals that apply to the user.
+                // Cookie will have all principals added to it.
+                AuthenticatorImpl ai = new AuthenticatorImpl();
+                Subject userSubject = new Subject();
+                userSubject.getPrincipals().add(p);
+                ai.augmentSubject(userSubject);
+                Set<Principal> userPrincipals = userSubject.getPrincipals();
+
+                // domains have to be harvested here somehow.
+                // what is the default domain
```

Comment:
This comment probably doesn't apply any more?  That work is done in the cookie manager.